### DISCUSSION TITLE
Trick to work around self deleting widgets.

### DISCUSSION THREAD

**Discussion by akhudek** [fltk-rs/fltk-rs#1398] [General]
In the application I'm developing I commonly have a group of widgets that I need to periodically rebuild based on data changes. To do this I do something like:

```rust
fn rebuild_group() {
  group.clear();
  group.begin();
  build_widgets();
  group.end();
}
```

Now image you have `build_widgets()` that looks like:

```rust
fn build_widgets(state: &State) {
  let mut button = Button::default().with_label("Update");
  button.set_callback({
    let mut state = state.Clone();
    move |_| {
      state.some_update();
      state.rebuild_group();
    }
  });
}
```

The above will produce a nasty failure because the rebuild code is deleting the object that the calling callback is attached to. You could work around this with a local message passing system and have the parent call the rebuild code, but that's a lot of boilerplate. 

One way I've found to work around this issue is to put the above rebuild code into zero length timeout call. E.g.

```rust

app::add_timeout3(0.0,{
  let mut state = state.clone();
  move |_| state.rebuild_group();
}); 
```

**Comment by MoAlyousef** [comment]
Indeed that is a problem.
Thanks for sharing this and is a neat use case of timers.
